gitlab/client.py

```diff
@@ -774,47 +774,51 @@ def http_request(
 
                 raise
 
-            self._check_redirects(result)
+            self._check_redirects(result.response)
 
-            if 200 <= result.status_code < 300:
-                return result
+            if 200 <= result.response.status_code < 300:
+                return result.response
```

Comment:
This is getting a bit nested here, could we maybe just define the variables so we don't repeat the usage of `result.response.attribute`? I think this way we might also be able to hide the backend details a bit later.

```python
headers = result.response.headers
status_code = result.response.status_code
content = result.response.content # replaces error_message = result.content
```

Reply:
Good idea. I've implemented it a bit differently. WDYT?

Reply:
Thanks, this works for me although I think we will need to add more Response attributes later, for history and reason phrases at least.